SPRK-377:fixed character limit

Mazharcodemaster opened the pull request:
[SPRK-377](https://etlonline.atlassian.net/jira/software/projects/SPRK/boards/35?selectedIssue=SPRK-377)

**Description**

The system allows users to enter excessively long text in the First Name and Last Name fields during both profile creation and profile update/edit. This can cause layout issues (e.g., button misalignment) in connected user cards and other UI sections where user details are displayed. Proper character limits are missing.

**Steps to Reproduce:**
1. Navigate to the Profile Creation or Edit Profile page.
2. Enter a very long text string (e.g., 200+ characters) in the First Name and Last Name fields.
3. Save the profile.
4. Navigate to any section where user details are displayed (e.g., Connections → Profile Activity).
5. Observe how the UI breaks or becomes misaligned.

**Actual Result:**
The system accepts long values for First and Last Name, which cause UI misalignment and layout issues in connected user cards and other sections.

**Expected Result:**
The system should enforce a reasonable character limit (e.g., 30–50 characters) for First Name and Last Name fields to prevent UI distortion. A validation error should be displayed if the input exceeds the limit.

**Implementation Details:**
1. Add input validation to both First Name and Last Name fields.
2. Restrict maximum character length to 30 characters.
3. Display a validation error message if the user exceeds this limit:

Mazharcodemaster committed:
SPRK-377:fixed character limit

Mazharcodemaster committed:
SPRK-377:implement zod validation in edit profile modal

Mazharcodemaster commented:
**1. Added Zod validation for Edit Profile Modal**

- Implemented robust schema-based validation using Zod.
- Ensures all profile fields (first name, last name, bio, skills, interests) are properly validated with clear error messages.

**2. Fixed Education Modal Year Validation**

- Resolved issues with handling start and end year fields in both “Create” and “Edit Education” modals.
- Empty year fields now correctly show “Year required” instead of “Invalid year”.
- Added real-time validation updates when users type — fields now revalidate immediately for a smoother experience.
- Prevented unnecessary error messages when the end year is empty (no more “start year must be before end year” if end year isn’t entered).

Usamavirk444 commented on src/components/Dashboard/profile/edit-profile-modal.tsx:
```diff
@@ -28,20 +28,31 @@ import { useAuthUser } from "@/src/hooks/useAuthUser"
+import z from "zod"
+
+const editProfileSchema = z.object({
+  first_name: z.string().min(1, "First name required").max(30, "Too long"),
```
correct the message here, and in last name and bio too

Mazharcodemaster committed:
SPRK-377: update character message

Mazharcodemaster commented:
SPRK-377:update character message

Mazharcodemaster committed:
SPRK-377: updated

Mazharcodemaster commented:
Fixed a runtime error occurring on the Super Admin side when opening the Edit Education modal due to missing profile data (null profile). Added safe handling for undefined profile objects.
